### PR TITLE
Backstory questionaire

### PR DESCRIPTION
**Pull request by Jordan-Renaud**
# Backstory Component with Questions

- currently just logs out the response

image

### CODE REVIEW

File: components/BackstoryQuestionaire.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    newAnswers[currentQuestionIndex] = answer;
+    setOwnChoiceAnswers(newAnswers);
+  }
+
+  function handleNavigationButtonClick(e) {
+    e.preventDefault();
+
+    e.target.value === "next"
+      ? setCurrentQuestionIndex(currentQuestionIndex + 1)
+      : setCurrentQuestionIndex(currentQuestionIndex - 1);
```
Comment (Darth-Knoppix) [inline components/BackstoryQuestionaire.js:80]: Another way to write this is:
```suggestion
setCurrentQuestionIndex(currentQuestionIndex + e.target.value === "next" ? 1 : -1)
```

Whichever you find more readable.

File: components/Question.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    setAnswers(target.value);
+  }
+
+  return (
+    <section>
+      <label htmlFor={`Question ${index}`}>{question.question}</label>
+      <ul>
+        {question.choices.map((answer, index) => (
+          <li key={index}>
```
Comment (Darth-Knoppix) [inline components/Question.js:23]: Ideally keys should be based on properties that can be tied to the thing you're iterating through, so in this case, I'd opt for `answer`
```suggestion
          <li key={answer}>
```

File: components/Question.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          <input
+            checked={answers.includes("Own choice")}
+            type="checkbox"
+            id="Own choice"
+            name="Own choice"
+            value="Own choice"
+            onChange={handleChoiceSelection}
+            disabled={
+              !answers.includes("Own choice") &&
+              answers.length === question.choiceAmountRequired
```
Comment (Darth-Knoppix) [inline components/Question.js:52]: This is used a couple of times, this could be moved to a variable within the loop

**Review by Darth-Knoppix — COMMENTED** [review]
Looking good, just a few suggestions

File: components/BackstoryQuestionaire.js (hunk 1 above)
Comment (Jordan-Renaud) [reply components/BackstoryQuestionaire.js:80]: Updating this breaks it completely for some reason, not sure why

**Review by Darth-Knoppix — APPROVED** [review]

**Pull request merged**